index.js

```diff
@@ -1,28 +1,19 @@
 'use strict';
 var url = require('url');
 var got = require('got');
-var registryUrl = require('registry-url');
-var rc = require('rc');
+var registryUrl = require('registry-auth-token/registry-url');
+var registryAuthToken = require('registry-auth-token');
 var semver = require('semver');
 
 module.exports = function (name, version) {
 	var scope = name.split('/')[0];
-	var pkgUrl = url.resolve(registryUrl(scope), encodeURIComponent(name).replace(/^%40/, '@'));
-	var npmrc = rc('npm');
-
-	var token;
-	if (!npmrc.registry || url.parse(npmrc.registry).hostname === 'registry.npmjs.org') {
-		token = npmrc[scope + ':_authToken'] || npmrc['//registry.npmjs.org/:_authToken'];
-	}
-
+	var regUrl = registryUrl(scope);
+	var pkgUrl = url.resolve(regUrl, encodeURIComponent(name).replace(/^%40/, '@'));
+	var authInfo = registryAuthToken(regUrl);
 	var headers = {};
 
-	if (token) {
-		if (process.env.NPM_TOKEN) {
```

Comment:
Actually, [it does](https://github.com/rexxars/registry-auth-token/blob/master/index.js#L65-L67). See [tests](https://github.com/rexxars/registry-auth-token/blob/master/test/auth-token.test.js#L71-L103).